mockgen/mockgen.go

```diff
@@ -225,6 +225,12 @@ func (g *generator) Generate(pkg *model.Package, pkgName string) error {
 		localNames[pkgName] = true
 	}
 
+	// Write a package-level documentation comment only if emitting a new
+	// package outright, which would not clobber the original's
+	// documentation.
+	if inPkg, outPkg := pkg.Name, pkgName; inPkg != outPkg {
```

Comment:
Taking a comparative approach:

The ubiquitous goprotobuf [emits the package documentation arbitrarily without checking](https://github.com/golang/protobuf/blob/157d9c53be5810dd5a0fac4a467f7d5f400042ea/protoc-gen-go/generator/generator.go#L1249) whether such output would collide. 

Like Protocol Buffers, folks generally wouldn't mix GoMock outputs with production, hand-written packages.  I am inclined to say we just emit it always or keep the good faith check — imperfections and all.

Thoughts?